warehouse/query-core/src/main/java/datawave/query/util/AttributeComparator.java

```diff
@@ -0,0 +1,203 @@
+package datawave.query.util;
+
+import com.google.common.collect.Sets;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Attributes;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public final class AttributeComparator {
+    
+    // this is a utility and should never be instantiated
+    private AttributeComparator() {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * Performs a simple comparison on a set of Attribute's data and some parts of their metadata
```

Comment:
Be explicit about what is considered for  equality, this could be rewritten to `Compares an Attribute's data, column visibility, and timestamp`. Realistically we should also compare the metadata for reasons detailed below

Reply:
Updated in new commit